documentation/tutorial/07-deploy-ingress.md

```diff
@@ -0,0 +1,183 @@
+# Deploy Nginx Ingress Integrator
+
+This is part of the [Charmed Temporal Tutorial](./00-introduction.md). Please
+refer to this page for more information and the overview of the content.
+
+The Charmed Temporal K8s operator exposes its service ports using the
+[Nginx Ingress Integrator](https://charmhub.io/nginx-ingress-integrator)
+operator. You must first make sure to have an
+[Nginx Ingress Controller](https://docs.nginx.com/nginx-ingress-controller/)
+deployed.
+
+## Enable TLS
+
+To enable TLS connections, you must have a TLS certificate stored as a k8s
+secret (default name is "temporal-tls"). A self-signed certificate for
+development purposes can be created as follows:
+
+```bash
+# Generate private key
+openssl genrsa -out server.key 2048
+
+# Generate a certificate signing request
+openssl req -new -key server.key -out server.csr -subj "/CN=temporal-k8s"
+
+# Create self-signed certificate
+openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt -extfile <(printf "subjectAltName=DNS:temporal-k8s")
+
+# Create a k8s secret
+kubectl create secret tls temporal-tls --cert=server.crt --key=server.key
```

Comment:
Here we assume that the user is already in the `temporal-model` namespace, further down when describing the ingress we are specifying the namespace. I think it is worth being consistent across the two.